sljit_src/sljitProtExecAllocator.c

```diff
@@ -132,38 +128,36 @@ static SLJIT_INLINE int create_tempfile(void)
 	}
 #endif
 
-#ifdef P_tmpdir
-	len = (P_tmpdir != NULL) ? strlen(P_tmpdir) : 0;
-
-	if (len > 0 && len < sizeof(tmp_name)) {
-		strcpy(tmp_name, P_tmpdir);
-		tmp_name_len = len;
-	}
-	else {
-		strcpy(tmp_name, "/tmp");
-		tmp_name_len = 4;
-	}
-#else
-	strcpy(tmp_name, "/tmp");
-	tmp_name_len = 4;
-#endif
-
 	dir = secure_getenv("TMPDIR");
 
 	if (dir) {
 		len = strlen(dir);
 		if (len > 0 && len < sizeof(tmp_name)) {
-			strcpy(tmp_name, dir);
+			if ((stat(dir, &st) == 0) && S_ISDIR(st.st_mode)) {
```

Comment:
Is this new code disallows passing an invalid directory to silently disabling JIT compiling?

Reply:
didn't consider that a feature, but yes if TMPDIR points to an invalid directory then we will fallback to use the other 2 options instead of failing.

I was considering this instead to be a reliability fix, so that something as important as the memory needed for JIT will be most likely to succeed than before.

IMHO, if we want to provide some sort of external flag to disable JIT we should define either a specific ENV variable or some other flag than a broken TMPDIR

Reply:
is this a feature (runtime disabling of JIT) worth considering?, I can see some uses of it (ex: in git where a JIT enabled pcre being used by `git grep` might be problematic, but that is just because git (at least for PCRE2) uses the JIT fastpath from PCRE which doesn't provide the normal fallback.

are there any other users of sljit where runtime disabling of JIT might be needed?, this and the paranoid mode you suggested might be even somehow related at least on use cases